AdobeMediaGallery/Model/Asset/Command/GetById.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AdobeMediaGallery\Model\Asset\Command;
+
+use Magento\AdobeMediaGalleryApi\Api\Data\AssetInterface;
+use Magento\AdobeMediaGalleryApi\Api\Data\AssetInterfaceFactory;
+use Magento\AdobeMediaGalleryApi\Model\Asset\Command\GetByIdInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Class GetById
+ */
+class GetById implements GetByIdInterface
+{
+    private const TABLE_ADOBE_MEDIA_GALLERY = 'adobe_media_gallery';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var AssetInterface
+     */
+    private $assetFactory;
+
+    /**
+     * GetAssetById constructor.
+     *
+     * @param ResourceConnection    $resourceConnection
+     * @param AssetInterfaceFactory $assetFactory
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        AssetInterfaceFactory $assetFactory
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->assetFactory = $assetFactory;
+    }
+
+    /**
+     * Get media asset.
+     *
+     * @param int $assetId
+     *
+     * @return AssetInterface
+     * @throws \Zend_Db_Statement_Exception
+     * @throws NoSuchEntityException
+     */
+    public function execute(int $assetId): AssetInterface
+    {
+        $connection = $this->resourceConnection->getConnection();
+
+        $select = $connection->select()
+            ->from(['amg' => self::TABLE_ADOBE_MEDIA_GALLERY])
+            ->where('amg.id = ?', $assetId);
+        $data = $connection->query($select)->fetchAll();
+
+        if (empty($data)) {
+            $message = __('There is no such media asset with "%1"', $assetId);
+            throw new NoSuchEntityException($message);
+        }
+
+        foreach ($data as $assetData) {
+            /** @var AssetInterface $asset */
+            $asset = $this->assetFactory->create();
+            $asset->setId($assetData[AssetInterface::ID]);
```

Comment:
`\Magento\Framework\EntityManager\HydratorInterface::hydrate`can be used here to simplify the logic and make it more flexible.

The only thing that is needed to be able to hydrate entity is adding the short configuration to di.xm i.e.:

    <type name="Magento\Framework\EntityManager\MetadataPool">
        <arguments>
            <argument name="metadata" xsi:type="array">
                <item name="Magento\AdobeStockAssetApi\Api\Data\AssetInterface" xsi:type="array">
                    <item name="entityTableName" xsi:type="string">adobe_stock_asset</item>
                    <item name="identifierField" xsi:type="string">id</item>
                </item>
            </argument>
        </arguments>
    </type>

Reply:
Good point! But I do not see the `@api` notation for the `Magento\Framework\EntityManager\HydratorInterface`. Are we in safety to use it for this new media gallery module?